compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java

```diff
@@ -384,8 +384,8 @@ private BuildOptions setBuildOptions(TomlTableNode tomlTableNode) {
 
         BuildOptions.BuildOptionsBuilder buildOptionsBuilder = BuildOptions.builder();
 
-        Boolean skipTests =
-                getBooleanFromBuildOptionsTableNode(tableNode, BuildOptions.OptionName.SKIP_TESTS.toString());
+        Boolean withTests = toBooleanFalseIfNull(
```

Comment:
Why do we need to introduce a new method here? Handling the null is done in the BuildOption getters. I think we can use `getBooleanFromBuildOptionsTableNode`

Reply:
I made this change because we still make use of the old setSkipTests function. 
https://github.com/ballerina-platform/ballerina-lang/pull/33714/files#diff-7e5ce460862aa48a7763a0055e868a61f7ba3af7745dccbff6a2ced886f04440R414

This is not an idea way of inverting the read boolean and passing it to setSkipTests(). What we are basically doing is, if `--with-tests` flag is used , or the build option is set, we are implying that `skipTests` is 'false'. Which is why null doesnt work in this case.

Reply:
I thik the better way of doing this would be to completely remove all instances of `skipTests` and replace it with `withTests` instead of juggling this boolean around